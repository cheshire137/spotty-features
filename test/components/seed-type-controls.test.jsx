import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import SeedTypeControls from '../../src/components/seed-type-controls.jsx'

describe('SeedTypeControls', () => {
  let component = null
  let seedType = 'track'

  const onSeedTypeChange = newType => {
    seedType = newType
  }

  beforeEach(() => {
    component = (
      <SeedTypeControls
        onSeedTypeChange={onSeedTypeChange}
        seedType={seedType}
      />
    )
  })

  afterEach(() => {
    seedType = 'track'
  })

  test('matches snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('reports when seed type changes', () => {
    const wrapper = shallow(component)
    const trackRadio = wrapper.find('input[value="track"]')
    expect(trackRadio.props().checked).toBe(true)

    const artistRadio = wrapper.find('input[value="artist"]')
    artistRadio.simulate('change', { target: { value: 'artist' } })

    expect(seedType).toBe('artist')
  })
})
