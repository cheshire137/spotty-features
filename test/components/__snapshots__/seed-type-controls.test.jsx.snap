exports[`SeedTypeControls matches snapshot 1`] = `
<div
  className="control">
  <label
    className="radio is-large">
    <input
      checked={true}
      name="seed-type"
      onChange={[Function]}
      type="radio"
      value="track" />
    Songs
  </label>
  <label
    className="radio is-large">
    <input
      checked={false}
      name="seed-type"
      onChange={[Function]}
      type="radio"
      value="artist" />
    Artists
  </label>
</div>
`;
